Swift Build warnings in 4.1

### Expected behavior
Build to complete successfully without any warnings

### Actual behavior

```
App/.build/checkouts/swift-nio.git-3108475404973543938/Sources/NIO/System.swift:60:112: warning: using '!' in this location is deprecated and will be removed in a future release; consider changing this to '?' instead
private let sysInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutablePointer<CChar>?, socklen_t) -> UnsafePointer<CChar>! = inet_ntop
                                                                                                               ^                   ~
                                                                                                                                   ?
```

### Steps to reproduce

1. Build NIO in Xcode 9.3 Beta 4

### SwiftNIO version/commit hash

1.3.0

### Swift & OS version (output of `swift --version && uname -a`)

```
$  swift --version && uname -a
Apple Swift version 4.1 (swiftlang-902.0.43 clang-902.0.37.1)
Target: x86_64-apple-darwin17.4.0
Darwin Tims-MBP.home 17.4.0 Darwin Kernel Version 17.4.0: Sun Dec 17 09:19:54 PST 2017; root:xnu-4570.41.2~1/RELEASE_X86_64 x86_64
```

--- a/Sources/NIO/System.swift
+++ b/Sources/NIO/System.swift
@@ -57,7 +57,7 @@ private let sysFreeifaddrs: @convention(c) (UnsafeMutablePointer<ifaddrs>?) -> V
 private let sysAF_INET = AF_INET
 private let sysAF_INET6 = AF_INET6
 private let sysAF_UNIX = AF_UNIX
-private let sysInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutablePointer<CChar>?, socklen_t) -> UnsafePointer<CChar>! = inet_ntop
+private let sysInet_ntop: @convention(c) (CInt, UnsafeRawPointer?, UnsafeMutablePointer<CChar>?, socklen_t) -> UnsafePointer<CChar>? = inet_ntop
 
 #if os(Linux)
 private let sysSendMmsg: @convention(c) (CInt, UnsafeMutablePointer<CNIOLinux_mmsghdr>?, CUnsignedInt, CInt) -> CInt = CNIOLinux_sendmmsg

--- a/Sources/NIOWebSocketServer/main.swift
+++ b/Sources/NIOWebSocketServer/main.swift
@@ -145,7 +145,7 @@ private final class WebSocketTimeHandler: ChannelInboundHandler {
         buffer.write(string: "\(theTime)")
 
         let frame = WebSocketFrame(fin: true, opcode: .text, data: buffer)
-        ctx.writeAndFlush(self.wrapOutboundOut(frame)).map { (_: Void) in
+        ctx.writeAndFlush(self.wrapOutboundOut(frame)).map { _ in
             _ = ctx.eventLoop.scheduleTask(in: .seconds(1), { self.sendTime(ctx: ctx) })
         }.whenFailure { (_: Error) in
             ctx.close(promise: nil)

--- a/Sources/NIOWebSocketServer/main.swift
+++ b/Sources/NIOWebSocketServer/main.swift
@@ -145,7 +145,7 @@ private final class WebSocketTimeHandler: ChannelInboundHandler {
         buffer.write(string: "\(theTime)")
 
         let frame = WebSocketFrame(fin: true, opcode: .text, data: buffer)
-        ctx.writeAndFlush(self.wrapOutboundOut(frame)).map { _ in
+        ctx.writeAndFlush(self.wrapOutboundOut(frame)).map {
             _ = ctx.eventLoop.scheduleTask(in: .seconds(1), { self.sendTime(ctx: ctx) })
         }.whenFailure { (_: Error) in
             ctx.close(promise: nil)

